### ISSUE TITLE
Struct fields that start with "ass" makes render() panic

### ISSUE THREAD

**Issue by mkawia** [sunng87/handlebars-rust#189]
Think I've encountered a bug. If you have a `Struct` and want to Serialize to render a template and it has a field with a name starting with "ass" eg "ass" or "asset_account" it panics . To illustrate:

`extern crate handlebars;
extern crate serde_json;

#[macro_use]
extern crate serde_derive;

use handlebars::Handlebars;

fn main() {
    let source = "{{#each asset_accounts }} {{name}} {{/each~}}";
    let mut handlebars = Handlebars::new();

    handlebars
        .register_template_string("hello", source)
        .unwrap();

    let data = Data {
        asset_accounts: vec![
            Asset {
                name: "Peter".to_string(),
            },
            Asset {
                name: "Cleveland".to_string(),
            },
        ],
    };

    match handlebars.render("hello", &data) {
        Ok(output) => {
            println!("output is {}", output);
        }
        Err(err) => {
            println!("this fails {:?}", err);
        }
    };
}

#[derive(Serialize, Deserialize, Debug)]
struct Data {
    asset_accounts: Vec<Asset>,
}

#[derive(Serialize, Deserialize, Debug)]
struct Asset {
    name: String,
}`

**Comment by dikaiosune** [comment]
😅 

I'm not a contributor, but I took a second to look into this on the latest version. I think there may be a bug here, but it's not in render and it's not a problem with "ass" in the variable name AFAICT. `as` as the start will produce a panic in `register_template_string`:

```rust
let source = "{{#each as }} {{name}} {{/each~}}";
let mut handlebars = Handlebars::new();

handlebars
    .register_template_string("hello", source)
    .unwrap();
```

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: TemplateError { reason: InvalidSyntax, template_name: Some("hello"), line_no: Some(1), column_no: Some(9) }', src/libcore/result.rs:860:4
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

IIRC, handlebars loops allow aliasing a loop variable like `{{#each account as a}}...{{/each}}`, maybe there's a parsing problem that assumes an identifier starting with `as` is invalid as it's expected to be a keyword?

Searching around the repo yields: https://github.com/sunng87/handlebars-rust/blob/f2ba28262ff4010cf7111588454aaff70c911ee7/src/grammar.pest#L29-L31

I'm not sure how pest/PEG is supposed to work here, but it seems like saying a param is maybe being prefix-matched as "not 'as'"? Not sure.

**Comment by sunng87** [comment]
Yes, it seems pest will won't match any token starts with "as" with my rule. I just added a quick fix that change `"as"` to `"as "`.

**Comment by Harleqin** [comment]
I'm not sure about the intent here, but wouldn't this then fail if the "as" is followed by a TAB instead of a SPACE, for example if someone is trying to align parts vertically?

I wonder whether the path to more obvious correctness might be first to tokenize, then parse the tokens.

**Comment by hashtagfoundation** [comment]
The framework I built also uses the "as" syntax everywhere for contexts... it looks for any whitespace before and after the "as", including vertical whitespace.

**Comment by sunng87** [comment]
I understand. I'm looking for better solution to unmatch `as` from `reference` in pest. For now I'm not sure how to exclude some keywords from it.

**Comment by sunng87** [comment]
This should be fixed in 571a1483. Pest rule is added to filter keywords out. It works for various whitespaces.
